text/0000-expect-attribute-count-parameter.md

```diff
@@ -0,0 +1,188 @@
+out how future features interact with this one.
+
+Nothing would be forced upon codebases, so there is no threat of increasing the
+complexity of learning the language to newcomers.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Add to `#[allow]` instead of `#[expect]`
+
+This could be made an option on the `#[allow]` attribute and entirely supplant the
+need for `#[expect]`. However, I think it makes more sense to include this on
+the `#[expect]` if such an attribute exists, and removing the `#[expect]`
+attribtue for just allowing a specific count would remove the case where you
+don't care how many violations exist.
+
+## Other ways of handling the count
+
+The `count` as proposed must match exactly the number of instances of the lint
+triggered inside. We could allow constructs like `count <= #`, either in
+addition to or instead of the currently-proposed exact match.
+
+In my view, this is worse for the case of gradually removing lints from an
+existing project. If an `expect` has 10 instances of the lint, I might slap an
+`#[expect(..., count <= 10)]` on it. Then, say I do some cleanup and reduce it
+down to 5 instances. Nothing prompts me to change the count, so I don't. Later,
+having forgotten about this, I do something else that adds back in up to 5
+instances of the lint, and the "slack" from the previous 5 being removed cause
+me to not get any warnings about the new instances.
+
+## Other ways of handling the inner lint instances
+
+If the `count` doesn't match, then we could continue to suppress the lints
+inside, but output one mega-lint which tells the user about all of the instances
+inside. However, I think that would be worse from a UX perspective.
+
+We could also track the level that was specified for this lint outside of the
+`#[expect]` annotation and reapply that level. However, this seems like extra
+work for marginal benefit, and it would lead to imo worse UX if the outside
+scope was an `#[allow]`, as it would render the lint locations effectively
+invisible.
+
+We could also require the `count` attribute be paired with a `level` attribute
+to indicate the level to pass through lints on a mismatched count, but in my
+opinion that adds too much complexity and makes it needlessly difficult to use.
+
+## Do Nothing
+
+We could do nothing, which would miss out on the benefits I outlined in the
+motivation section.
+
+# Prior art
+[prior-art]: #prior-art
+
+## Syntax in Rust
+
+Other attributes take such named parameters, such as the `stable` and `unstable`
+attributes, and the `reason` parameter proposed for all lint levels in the same
+RFC as the `#[expect]` lint level.
+
+## Behavior
+
+I am unaware of any languages which have a similar functionality to what I am
+describing here.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Can we allow `#[expect(..., count = 0)]` with some useful behavior? What about
```

Comment:
`expect(lint, count = 0)` seems like it should be the same as `deny(lint)`.

Reply:
Agreed. I could see `count = 0` being convenient for a tool that decreases the expected count when it is too high. Such a tool would then not have to deal with this edge case.

In my mind `expect(lint, count = 0)` should compile with a warning (or clippy lint) suggesting using `deny` instead.